### PR TITLE
JHUAPL buildings_to_dsm

### PR DESCRIPTION
**Pull request by jacobderosa**
Changes made to buildings_to_dsm.py by JHUAPL to allow for the replacement of P3D with Colmap and VisSat

### CODE REVIEW

**Comment by mleotta** [comment]
@danlipsa you would be a good candidate to review these changes.  I believe you wrote this code.  JHU/APL was having trouble with rendering the DSM from building meshes and proposed this change.

**Comment by mleotta** [comment]
@danlipsa could you look at this branch.  We could use your input on this since you wrote this originally.

File: tools/buildings_to_dsm.py [hunk 1]
```diff
@@ -17,7 +17,32 @@
...
+    in a Dockerized notebook, or over a server without X forwarding.
+
+    * Requires the following packages:
+      * `sudo apt-get install libgl1-mesa-dev xvfb`
+    '''
+
+    import subprocess
+    import vtk
+
+    vtk.OFFSCREEN = True
```
Comment (danlipsa) [inline tools/buildings_to_dsm.py:37]: Is there a variable like that? What does it do?

File: tools/buildings_to_dsm.py [hunk 2]
```diff
@@ -181,6 +207,7 @@ def main(args):
     ren = vtk.vtkRenderer()
     renWin = vtk.vtkRenderWindow()
     renWin.OffScreenRenderingOn()
+    renWin.SetOffScreenRendering(1)
```
Comment (danlipsa) [inline tools/buildings_to_dsm.py:210]: The previous like should do the same thing.

File: tools/buildings_to_dsm.py [hunk 3]
```diff
@@ -277,56 +304,61 @@ def main(args):
...
+                dsmElevation = elevation
+            else:
+                # elevation has nans in places other than buildings
+                dsmElevation = numpy.fmax(dtmRaster, elevation)
+            dsm.GetRasterBand(1).WriteArray(dsmElevation)
+            if nodata:
+                dsm.GetRasterBand(1).SetNoDataValue(nodata)
+
+        except:
+            pass
```
Comment (danlipsa) [inline tools/buildings_to_dsm.py:362]: Who throws the exception? Why pass and not print the error?

File: tools/buildings_to_dsm.py (hunk 1 above)
Comment (mleotta) [reply tools/buildings_to_dsm.py:37]: I can't find any reference to this variable.  Maybe we should remove this and confirm that it still works.

File: tools/buildings_to_dsm.py (hunk 3 above)
Comment (mleotta) [reply tools/buildings_to_dsm.py:362]: I wonder if this was just added for debugging before the offscreen rendering was fixed.  Maybe we can ignore this change.  It doesn't seem like a good idea to catch all exceptions and then pass without notification.

File: tools/buildings_to_dsm.py (hunk 1 above)
Comment (jacobderosa) [reply tools/buildings_to_dsm.py:37]: I'll rerun this step with the old version.

**Comment by mleotta** [comment]
These changes do not appear to be necessary

**Pull request closed without merging**